integration-tests/src/test/java/org/zowe/apiml/integration/ha/WebSocketChaoticTest.java

```diff
@@ -0,0 +1,160 @@
+        return client.doHandshake(appendResponseHandler(response, countToNotify), headers, uri).get(30000, TimeUnit.MILLISECONDS);
+    }
+
+    @Nested
+    class WhenRoutingSessionGivenHA {
+        @Nested
+        class OpeningASession {
+            @Nested
+            class WhenAnInstanceIsOff {
+                @ParameterizedTest(name = "WhenRoutingSessionGivenHA.OpeningASession.WhenAnInstanceIsOff#propagateTheSessionToTheAliveInstance {0}")
+                @ValueSource(strings = {"/discoverableclient/ws/v1/uppercase", "/ws/v1/discoverableclient/uppercase"})
+                void propagateTheSessionToTheAliveInstance(String path) throws Exception {
+                    final StringBuilder response = new StringBuilder();
+
+                    WebSocketSession session = appendingWebSocketSession(discoverableClientGatewayUrl(path, 0), VALID_AUTH_HEADERS, response, 1);
+
+                    // shutdown one instance of DC to check whether the message can reach out the other instance
+                    haDiscoverableClientRequests.shutdown(0);
+                    session.sendMessage(new TextMessage("hello world!"));
+                    synchronized (response) {
+                        response.wait(WAIT_TIMEOUT_MS);
+                    }
+
+                    assertEquals("HELLO WORLD!", response.toString());
+                    session.close();
+                }
+
+                @ParameterizedTest(name = "WhenRoutingSessionGivenHA.OpeningASession.WhenAnInstanceIsOff#newSessionCanBeCreated {0}")
+                @ValueSource(strings = {"/discoverableclient/ws/v1/uppercase", "/ws/v1/discoverableclient/uppercase"})
+                void newSessionCanBeCreated(String path) throws Exception {
+                    final StringBuilder response = new StringBuilder();
+
+                    // create websocket session using the second instance of Gateway
+                    WebSocketSession session = appendingWebSocketSession(discoverableClientGatewayUrl(path, 1), VALID_AUTH_HEADERS, response, 1);
+
+                    session.sendMessage(new TextMessage("hello world 2!"));
+                    synchronized (response) {
+                        response.wait(WAIT_TIMEOUT_MS);
+                    }
+
+                    assertEquals("HELLO WORLD 2!", response.toString());
+                    session.close();
+                }
+            }
+
+            @Nested
+            class WhenAGatewayInstanceIsOff {
+
+                @ParameterizedTest(name = "WhenRoutingSessionGivenHA.OpeningASession.WhenAGatewayInstanceIsOff#newSessionCanBeCreated {0}")
+                @ValueSource(strings = {"/discoverableclient/ws/v1/uppercase", "/ws/v1/discoverableclient/uppercase"})
+                void newSessionCanBeCreated(String path) throws Exception {
+                    final StringBuilder response = new StringBuilder();
+
+                    // take off an instance of Gateway
+                    haGatewayRequests.shutdown(0);
```

Comment:
Same concern as for the previous test.

Reply:
Solved both cases with a check on the `path` value, so shutdown is executed only once per test